ngraph/test/type_prop/proposal.cpp

```diff
@@ -143,6 +169,229 @@ TEST(type_prop, proposal_v0_shape_infer)
     ASSERT_EQ(op->get_output_shape(0), (Shape{batch_size * attrs.post_nms_topn, 5}));
 }
 
+TEST(type_prop, proposal_v0_dynamic_class_probs_dim1_batch_size_infer)
+{
+    op::ProposalAttrs attrs;
+    attrs.post_nms_topn = 1;
+    const size_t batch_size = 2;
+    auto class_probs =
+        make_shared<op::Parameter>(element::f32, PartialShape{Dimension::dynamic(), 2, 3, 4});
+    auto class_bbox_deltas = make_shared<op::Parameter>(element::f32, Shape{batch_size, 4, 3, 4});
+    auto image_shape = make_shared<op::Parameter>(element::f32, Shape{3});
+
+    auto op = make_shared<op::v0::Proposal>(class_probs, class_bbox_deltas, image_shape, attrs);
+    ASSERT_EQ(op->get_output_shape(0), (Shape{batch_size * attrs.post_nms_topn, 5}));
+}
+
+TEST(type_prop, proposal_v0_dynamic_bbox_deltas_dim1_batch_size_infer)
+{
+    op::ProposalAttrs attrs;
+    attrs.post_nms_topn = 1;
+    const size_t batch_size = 2;
+    auto class_probs =
+        make_shared<op::Parameter>(element::f32, PartialShape{Dimension::dynamic(), 2, 3, 4});
+    auto class_bbox_deltas = make_shared<op::Parameter>(element::f32, Shape{batch_size, 4, 3, 4});
+    auto image_shape = make_shared<op::Parameter>(element::f32, Shape{3});
+
+    auto op = make_shared<op::v0::Proposal>(class_probs, class_bbox_deltas, image_shape, attrs);
+    ASSERT_EQ(op->get_output_shape(0), (Shape{batch_size * attrs.post_nms_topn, 5}));
+}
+
+TEST(type_prop, proposal_v0_dynamic_class_probs_bbox_deltas_dim1_batch_size_infer)
+{
+    op::ProposalAttrs attrs;
+    auto class_probs =
+        make_shared<op::Parameter>(element::f32, PartialShape{Dimension::dynamic(), 2, 3, 4});
+    auto class_bbox_deltas =
+        make_shared<op::Parameter>(element::f32, PartialShape{Dimension::dynamic(), 4, 3, 4});
```

Comment:
Add one more test when both batch dimensions of `class_probs` and `class_bbox_deltas` have some range, for example, `[10, 14]` and `[8, 15]`. The output batch dimension should be `[8, 14]` in this case.

Reply:
In the example you gave the output dimension would be `[10, 14]`. I'll add one testcase each for v0 and v4 ops (to check also probabilities output shape in v4 case)

Reply:
Yes, you are right. My fault